internal/armada/configuration/types.go

```diff
@@ -100,6 +101,12 @@ type SchedulingConfig struct {
 	MinJobResources                           v1.ResourceList
 }
 
+type PreemptionConfig struct {
+	Enabled              bool
+	PriorityClasses      map[string]int32
```

Comment:
Should we just use the Kubernetes API for PriorityClasses?

Reply:
This is a mapping so Armada Server knows which Priority Classes are supported, and also to know the priorities so it can correctly calculate how much resources it can preempt for a prioritized job

Reply:
Maybe I wasn't clear.  I am asking if we should use the Kubernetes definition for PriorityClasses.  

https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#example-priorityclass

Reply:
Still not clear why, we are only interested in the Priority Class name (string) and the Priority value (int32)

Reply:
One argument in favor of Kevin's suggestion is that we may later want to support features added to Kubernetes priority classes. For example, we may want to mark a priority class as non-preemptive. However, I think for now we can leave it like this and then update the config later if necessary.